saleor/order/events.py

```diff
@@ -904,6 +904,8 @@ def order_line_product_removed_event(
     app: AppType,
     order_lines: List[Tuple[int, OrderLine]],
 ):
+    if not user_is_valid(user):
+        user = None
```

Comment:
How did we make it this far in code without a user? It looks like you're fixing bugs on the wrong layer.

Reply:
That's a good point :thinking:

Reply:
If `requestor` is an `App` then the `user` is `AnonymousUser`. Then if we try to pass `AnonymousUser` to `OrderEvent.user` field, it crashes and rightfully so.

Reply:
This check is present in some event functions, but not all of them. Shouldn't we always be checking this?

Reply:
We should, as we can get `AnonymousUser` there. But the better option will be not providing the `AnonymousUser` to the events, IMO we should fix the problem on the different layer as @patrys proposed.

Reply:
We should not be passing AnonymousUser that deep, it should be None when we don't have a user instance.

Reply:
@patrys You're right but this fixes the issue. Rejecting user earlier in the flow requires more work with refactoring and adjusting tests. I think we should use a band-aid (merge this PR) and then work on proper solution (which is what I'm currently doing anyway).